airstack-modules/modules/airstack/poap/poap.ts

```diff
@@ -0,0 +1,338 @@
+        token.save()
+
+        let ownerAttendee = getOrCreatePoapAttendee(chainId, owner.toHexString(), airBlock)
+        ownerAttendee.lastUpdatedIndex = updateAirEntityCounter(
+            AIR_POAP_ATTENDEE_UPDATED_INDEX_ID,
+            airBlock
+        )
+        ownerAttendee.tokensOwned = ownerAttendee.tokensOwned.plus(BIGINT_ONE)
+        ownerAttendee.save()
+
+        let event = getOrCreateAirPoapEvent(eventId, airBlock)
+        event.lastUpdatedIndex = updateAirEntityCounter(
+            AIR_POAP_EVENT_LAST_UPDATED_INDEX_ID,
+            airBlock
+        )
+        event.save()
+
+        // tokenMints = number of tokens minted under particular event
+        event.tokenMints = event.tokenMints.plus(BIGINT_ONE)
+        event.save()
+        let eventAttendee = getOrCreateAirPoapEventAttendee(
+            chainId,
+            tokenAddress,
+            airBlock,
+            tokenId,
+            tokenURI,
+            event,
+            ownerAttendee,
+            event.tokenMints // used as mintOrder
+        )
+        eventAttendee.save()
+
+        createAirPoapMintTransaction(
+            chainId,
+            airBlock,
+            transactionHash,
+            logOrCallIndex,
+            token,
+            tokenId,
+            eventAttendee,
+            ownerAttendee
+        )
+    }
+    /**
+     * @dev this function is used to track transfer transactions of POAPS
+     * @param block, ethereum block
+     * @param transactionHash, transaction hash
+     * @param logOrCallIndex, transaction log or call index
+     * @param eventId, event Id of POAP
+     * @param tokenid, token Id of POAP
+     * @param from, from address of token
+     * @param to, to address of token
+     */
+
+    export function trackPoapTransferTransactions(
+        block: ethereum.Block,
+        transactionHash: string,
+        logOrCallIndex: BigInt,
+        eventId: BigInt,
+        tokenId: BigInt,
+        from: Bytes,
+        to: Bytes
+    ): void {
+        const chainId = getChainId()
+
+        let airBlock = getOrCreateAirBlock(
+            chainId,
+            block.number,
+            block.hash.toHexString(),
+            block.timestamp
+        )
+        airBlock.save()
+
+        let eventAttendeeId = getAirPoapEventAttendeeId(eventId.toString(), tokenId)
+        let eventAttendee = AirPoapEventAttendee.load(eventAttendeeId)
+        if (eventAttendee == null) {
+            throw new Error("eventAttendee should not be null")
+        }
+
+        // creating / updating from attendee
+        let fromAttendee = getOrCreatePoapAttendee(chainId, from.toHexString(), airBlock)
+        fromAttendee.updatedAt = airBlock.id
+        fromAttendee.tokensOwned = fromAttendee.tokensOwned.minus(BIGINT_ONE)
+        fromAttendee.lastUpdatedIndex = updateAirEntityCounter(
+            AIR_POAP_ATTENDEE_UPDATED_INDEX_ID,
+            airBlock
+        )
+        fromAttendee.save()
```

Comment:
In general lets create functions for each updates & wrap logic for last updated index

Reply:
done